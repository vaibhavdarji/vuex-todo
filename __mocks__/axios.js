const axios = {
  get: jest.fn(() =>
    Promise.resolve({
      todos: []
    })
  )
};

export default axios;
