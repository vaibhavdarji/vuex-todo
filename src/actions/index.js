import axios from 'axios';
import {
  GET_TODOS_MUTATION,
  ADD_TODO_MUTATION,
  COMPLETE_TODO_MUTATION,
  REMOVE_TODO_MUTATION
} from '../constants';

const URL = 'http://localhost:5000/todos';
const config = { headers: { 'Content-Type': 'application/json' } };

const randomId = () =>
  Math.random()
    .toString()
    .substr(2, 10);

export const ADD_TODO = ({ commit }, todo) => {
  const newTodo = {
    value: todo,
    completed: false,
    editable: false,
    id: randomId()
  };
  axios.post(URL, newTodo).then(_ => commit('ADD_TODO_MUTATION', newTodo));
};

export const UPDATE_TODO = (context, todo) => {
  axios.put(`${URL}/${todo.id}`, todo, config).then(() => {
    context.commit('UPDATE_TODO_MUTATION', todo);
  });
};

export const COMPLETE_TODO = (context, todo) => {
  context.commit('COMPLETE_TODO_MUTATION', todo);
};

export const REMOVE_TODO = (context, todo) => {
  axios.delete(`${URL}/${todo.id}`).then(_ => {
    context.commit('REMOVE_TODO_MUTATION', todo);
  });
  // context.commit('REMOVE_TODO_MUTATION', todo);
};

export const GET_TODO = context => {
  context.commit('SET_LOADING', true);
  axios
    .get(URL)
    .then(response => response.data)
    .then(todos => {
      context.commit('SET_TODOS', todos);
      context.commit('SET_LOADING', false);
    });
};
