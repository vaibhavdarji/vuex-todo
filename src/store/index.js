import Vue from 'vue';
import Vuex from 'vuex';
import * as actions from '../actions';
import * as mutations from '../mutations';
Vue.use(Vuex);

const state = {
  loading: true,
  todos: []
};

const getters = {
  todos: state => state.todos,
  loading: state => state.loading
};

const store = new Vuex.Store({
  state,
  getters,
  actions,
  mutations
});

export default store;
