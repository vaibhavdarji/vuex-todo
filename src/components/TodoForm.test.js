import { mount, shallow, shallowMount } from 'vue-test-utils';
import store from '../store';
import TodoFrom from './TodoForm';
import { ADD_TODO } from '../actions';
import axios from 'axios';

describe('TodoForm.vue', () => {
  it('renders an input field', () => {
    const wrapper = shallow(TodoFrom);
    expect(wrapper.element).toMatchSnapshot();
    expect(wrapper.find('input')).toBeDefined();
  });

  // describe('add Todo', () => {
  //   it('adds todo from value of input field, if enter key is pressed', done => {
  //     const spy = jest.spyOn(store, 'dispatch');
  //     const givenValue = 'Eat Subway ';
  //     const wrapper = mount(TodoFrom, { store });
  //     const input = wrapper.find('.todo-input');
  //     input.element.value = givenValue;
  //     input.trigger('input');
  //     input.trigger('keyup.enter');

  //     wrapper.vm.$nextTick(() => {
  //       expect(spy).toHaveBeenCalledWith({
  //         title: givenValue.trim()
  //       });
  //       done();
  //     });
  //   });
  // });
});
