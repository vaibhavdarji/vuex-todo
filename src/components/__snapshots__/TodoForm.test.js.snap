// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoForm.vue renders an input field 1`] = `
<el-form
  class="todo-form"
  model="[object Object]"
  rules="[object Object]"
>
  <el-form-item
    prop="todoItem"
  >
    <el-col
      span="20"
    >
      <el-input
        class="todo-input"
        placeholder="Add Todo"
      />
    </el-col>
     
    <el-col
      push="3"
      span="2"
    >
      <el-button
        type="primary"
      >
        Add
      </el-button>
    </el-col>
  </el-form-item>
</el-form>
`;
